frontend/src/components/elements/CodeBlock/styled-components.ts

```diff
@@ -105,50 +105,63 @@ export const StyledPre = styled.pre(({ theme }) => ({
   },
 }))
 
-export const StyledCopyButton = styled.button(({ theme }) => ({
-  opacity: 0,
-  height: "2.5rem",
-  padding: 0,
-  width: "2.5rem",
-  transition: "opacity 300ms 150ms, transform 300ms 150ms",
-  border: "none",
-  backgroundColor: theme.colors.transparent,
-  color: theme.colors.fadedText60,
-  borderRadius: theme.radii.xl,
-  transform: "scale(0)",
-
-  "&:active, &:focus, &:hover": {
-    opacity: 1,
-    transform: "scale(1)",
-    outline: "none",
-    color: theme.colors.bodyText,
-    transition: "none",
-  },
-}))
-
 export const StyledCopyButtonContainer = styled.div(({ theme }) => ({
+  opacity: 0,
   padding: `${theme.spacing.sm} ${theme.spacing.sm} 0 0`,
   top: 0,
   right: 0,
   position: "absolute",
-  width: "2.75rem",
+  width: "100%",
   height: "100%",
   backgroundColor: theme.colors.transparent,
   zIndex: theme.zIndices.sidebar + 1,
+  display: "flex",
+  justifyContent: "flex-end",
+  alignItems: "flex-start",
+  transition: "opacity 300ms 150ms",
+  pointerEvents: "none",
 }))
 
 export const StyledCodeBlock = styled.div(({ theme }) => ({
   position: "relative",
   marginLeft: theme.spacing.none,
   marginRight: theme.spacing.none,
   marginTop: theme.spacing.none,
-  marginBottom: theme.spacing.lg,
+  "&:not(:last-child)": {
+    marginBottom: theme.spacing.lg,
+  },
+
   "&:hover": {
-    [StyledCopyButton as any]: {
+    [`${StyledCopyButtonContainer}`]: {
       opacity: 1,
-      transform: "scale(1)",
-      outline: "none",
-      transition: "none",
     },
   },
 }))
+
+export const StyledCopyButton = styled.button(({ theme }) => ({
+  pointerEvents: "auto",
+  height: "2.5rem",
+  padding: 0,
+  width: "2.5rem",
+  border: "none",
+  backgroundColor: theme.colors.transparent,
+  color: theme.colors.fadedText60,
+  borderRadius: theme.radii.xl,
+  transform: "scale(0)",
+
+  [`${StyledCodeBlock}:hover &`]: {
+    opacity: 1,
+    transform: "scale(1)",
+    outline: "none",
+    transition: "none",
+    color: theme.colors.bodyText,
+  },
+
+  "&:active, &:focus, &:hover": {
```

Comment:
I wonder why, we have two code blocks with identical styles.

Reply:
Which two code blocks are you referring to?

Reply:
- `StyledPre` is responsible for the styling of the code itself, including various tokens and colors.
- `StyledCopyButtonContainer` sets up the container for the copy button, positioning it correctly within the code block and handling hover states.
- `StyledCodeBlock` provides the necessary relative positioning and margins for the code block.
- `StyledCopyButton` is the styling for the copy button itself, handling its appearance and hover states (sets the position, size, color, and various styles to show and hide the button on hover and click).

Each of these styled components serves a unique purpose IIUC